src/api.rs

```diff
@@ -510,8 +510,10 @@ pub struct Context<T: Pixel> {
 
 #[derive(Clone, Copy, Debug)]
 pub enum EncoderStatus {
-  /// The encoder needs more Frames to produce an output Packet
+  /// The encoder needs more data to produce an output Packet--used with frame reordering
   NeedMoreData,
+  /// The encoder needs more Frames to analyze lookahead
+  NeedMoreFrames,
```

Comment:
They are the same condition: The encoding cannot continue until more data is provided.

Reply:
Technically, not quite. In one case we have the data in the frame queue, and just need to increment the index to read it. In the other case, the data is not yet in the frame queue.

Reply:
That implementation detail should not leak to the user IMHO.

Reply:
I agree, but I don't know of a better way to implement this without making the Context aware of the decoder. We've explicitly chosen to keep the Decoder trait and anything implementing it as part of the CLI.

I'll see if I can think over it and come up with a better approach.

Reply:
Let's unbreak at least the experience for the user first. Land this patch.

Reply:
I'll try to see what happens when you use only `NeedMoreData` myself later.